pipeline/train/train.py

```diff
@@ -0,0 +1,413 @@
+          - path/to/corpus.en.zst
+          - path/to/corpus.fr.zst
+          - path/to/corpus.aln.zst
+
+        And then builds:
+          - path/to/corpus.enfr.tsv
+    """
+    src_path = Path(f"{dataset_prefix}.{src}.zst")
+    trg_path = Path(f"{dataset_prefix}.{trg}.zst")
+    # OpusTrainer supports only tsv and gzip
+    # TODO: pigz is not installed on the generic Taskcluster worker, so we use datasets in decompressed mode for now
+    tsv_path = Path(f"{dataset_prefix}.{src}{trg}.tsv")  # .gz
+
+    with ExitStack() as stack:
+        tsv_outfile = stack.enter_context(write_lines(tsv_path))
+        src_lines: Generator[str, Any, Any] = stack.enter_context(read_lines(src_path))
+        trg_lines: Generator[str, Any, Any] = stack.enter_context(read_lines(trg_path))
+
+        logger.info(f"Generating tsv dataset: {tsv_path}")
+
+        if alignments_file:
+            logger.info(f"Using alignments file: {alignments_file}")
+
+            aln_lines: Generator[str, Any, Any] = stack.enter_context(
+                read_lines(f"{alignments_file}")
+            )
+            empty_alignments = []
+
+            for src_line, trg_line, aln_line in zip(src_lines, trg_lines, aln_lines):
+                if not aln_line:
+                    empty_alignments.append((src_line, trg_line))
+                    continue
+                tsv_outfile.write(f"{src_line.strip()}\t{trg_line.strip()}\t{aln_line.strip()}\n")
+
+            if empty_alignments:
+                logger.info(f"Number of empty alignments for {len(alignments_file)}")
+                logger.info("Sample of empty alignments:")
+                random.shuffle(empty_alignments)
+                for src_line, trg_line in empty_alignments[:50]:
+                    logger.info(f"  src: {src_line.strip()}")
+                    logger.info(f"  trg: {trg_line.strip()}")
+
+        else:
+            for src_line, trg_line in zip(src_lines, trg_lines):
+                tsv_outfile.write(f"{src_line.strip()}\t{trg_line.strip()}\n")
+
+    logger.info("Freeing up disk space after TSV merge.")
+    logger.info(f"Removing {src_path}")
+    src_path.unlink()
+    logger.info(f"Removing {trg_path}")
+    trg_path.unlink()
+    if alignments_file:
+        logger.info(f"Removing {alignments_file}")
+        alignments_file.unlink()
+
+    return tsv_path
+
+
+def get_log_parser_command():
+    if shutil.which("parse_tc_logs") is None:
+        logger.info("Weight & Biases publication script is not available.")
+        return ["cat"]
+
+    if "TEST_ARTIFACTS" in os.environ:
+        logger.info("Weight & Biases publication is disabled for unit tests.")
+        return ["cat"]
+
+    logger.info("Weight & Biases publication is available.")
+    return ["parse_tc_logs", "--from-stream", "--publish-group-logs", "-v"]
+
+
+class TrainCLI:
+    def __init__(self, args: Any, temp_dir: Path) -> None:
+        self.temp_dir = temp_dir
+        self.vocab: Path = args.vocab
```

Comment:
It might be worth moving the args to a separate data class and do arg parsing there. It will be easier to work with, log and extend (optional suggestion)

Reply:
I looked at doing this, but my original intent was for the TrainCLI to be the dataclass. Since this is optional, I'll pass on it for now.